internal/lintcontext/context.go

```diff
@@ -23,22 +23,37 @@ type InvalidObject struct {
 }
 
 // A LintContext represents the context for a lint run.
-type LintContext struct {
+type LintContext interface {
+	GetObjects() []Object
+	GetInvalidObjects() []InvalidObject
+}
+
+type lintContextImpl struct {
 	objects        []Object
 	invalidObjects []InvalidObject
 }
 
 // Objects returns the (valid) objects loaded from this LintContext.
-func (l *LintContext) Objects() []Object {
+func (l *lintContextImpl) GetObjects() []Object {
 	return l.objects
 }
 
+// AddObject adds a valid object to this LintContext
+func (l *lintContextImpl) addObjects(objs ...Object) {
+	l.objects = append(l.objects, objs...)
+}
+
 // InvalidObjects returns any objects that we attempted to load, but which were invalid.
-func (l *LintContext) InvalidObjects() []InvalidObject {
+func (l *lintContextImpl) GetInvalidObjects() []InvalidObject {
 	return l.invalidObjects
 }
 
+// AddInvalidObject adds an invalid object to this LintContext
+func (l *lintContextImpl) addInvalidObjects(objs ...InvalidObject) {
+	l.invalidObjects = append(l.invalidObjects, objs...)
+}
+
 // New returns a ready-to-use, empty, lint context.
-func New() *LintContext {
-	return &LintContext{}
+func New() *lintContextImpl {
```

Comment:
```suggestion
func New() LintContext {
```

Reply:
So I actually tried making it LintContext, but then `CreateContexts` also access functions like `loadObjectsFromReader` which is only available to lintContectImpl. I made sure the return values from `CreateContexts` are `[]LintContext`, but wasn't sure if there is another route that I can take here.

Reply:
Ahh, that makes sense. The right answer in this case is to unexport the `New` function. (Initially, it was used by outside callers, but after teh `CreateContexts` refactor it doesn't seen required any more).

Reply:
ah yep yep that is true. should just unexport it. will do